plugins/obs-qsv11/QSV_Encoder_Internal.cpp

```diff
@@ -257,110 +256,114 @@ mfxStatus QSV_Encoder_Internal::InitParams(qsv_param_t *pParams,
 
 	m_mfxEncParams.mfx.RateControlMethod = pParams->nRateControl;
 
 	switch (pParams->nRateControl) {
 	case MFX_RATECONTROL_CBR:
 		m_mfxEncParams.mfx.TargetKbps = pParams->nTargetBitRate;
-		m_mfxEncParams.mfx.BufferSizeInKB = pParams->nTargetBitRate * 2;
+		m_mfxEncParams.mfx.BufferSizeInKB =
+			(pParams->nTargetBitRate / 8) * 2;
 		m_mfxEncParams.mfx.InitialDelayInKB =
-			pParams->nTargetBitRate * 1;
+			(pParams->nTargetBitRate / 8) * 1;
 		break;
 	case MFX_RATECONTROL_VBR:
-	case MFX_RATECONTROL_VCM:
 		m_mfxEncParams.mfx.TargetKbps = pParams->nTargetBitRate;
 		m_mfxEncParams.mfx.MaxKbps = pParams->nMaxBitRate;
-		m_mfxEncParams.mfx.BufferSizeInKB = pParams->nTargetBitRate * 2;
+		m_mfxEncParams.mfx.BufferSizeInKB =
+			(pParams->nTargetBitRate / 8) * 2;
 		m_mfxEncParams.mfx.InitialDelayInKB =
-			pParams->nTargetBitRate * 1;
+			(pParams->nTargetBitRate / 8) * 1;
 		break;
 	case MFX_RATECONTROL_CQP:
 		m_mfxEncParams.mfx.QPI = pParams->nQPI;
 		m_mfxEncParams.mfx.QPB = pParams->nQPB;
 		m_mfxEncParams.mfx.QPP = pParams->nQPP;
 		break;
-	case MFX_RATECONTROL_AVBR:
-		m_mfxEncParams.mfx.TargetKbps = pParams->nTargetBitRate;
-		m_mfxEncParams.mfx.Accuracy = pParams->nAccuracy;
-		m_mfxEncParams.mfx.Convergence = pParams->nConvergence;
-		break;
 	case MFX_RATECONTROL_ICQ:
 		m_mfxEncParams.mfx.ICQQuality = pParams->nICQQuality;
 		break;
-	case MFX_RATECONTROL_LA:
-		m_mfxEncParams.mfx.TargetKbps = pParams->nTargetBitRate;
-		break;
-	case MFX_RATECONTROL_LA_ICQ:
-		m_mfxEncParams.mfx.ICQQuality = pParams->nICQQuality;
-		break;
-	case MFX_RATECONTROL_LA_HRD:
-		m_mfxEncParams.mfx.TargetKbps = pParams->nTargetBitRate;
-		m_mfxEncParams.mfx.MaxKbps = pParams->nTargetBitRate;
-		break;
 	default:
 		break;
 	}
 
 	m_mfxEncParams.AsyncDepth = pParams->nAsyncDepth;
 	m_mfxEncParams.mfx.GopPicSize =
-		(mfxU16)(pParams->nKeyIntSec * pParams->nFpsNum /
-			 (float)pParams->nFpsDen);
-
-	if (m_ver.Major == 1 && m_ver.Minor >= 8) {
-		memset(&m_co2, 0, sizeof(mfxExtCodingOption2));
-		m_co2.Header.BufferId = MFX_EXTBUFF_CODING_OPTION2;
-		m_co2.Header.BufferSz = sizeof(m_co2);
-		if (codec != QSV_CODEC_AVC)
-			m_co2.RepeatPPS = MFX_CODINGOPTION_ON;
-		if (pParams->nRateControl == MFX_RATECONTROL_LA_ICQ ||
-		    pParams->nRateControl == MFX_RATECONTROL_LA)
+		(pParams->nKeyIntSec)
+			? (mfxU16)(pParams->nKeyIntSec * pParams->nFpsNum /
+				   (float)pParams->nFpsDen)
+			: 240;
+
+	memset(&m_co2, 0, sizeof(mfxExtCodingOption2));
+	m_co2.Header.BufferId = MFX_EXTBUFF_CODING_OPTION2;
+	m_co2.Header.BufferSz = sizeof(m_co2);
+	if (pParams->bRepeatHeaders)
+		m_co2.RepeatPPS = MFX_CODINGOPTION_ON;
+	else
+		m_co2.RepeatPPS = MFX_CODINGOPTION_OFF;
+
+	memset(&m_co3, 0, sizeof(mfxExtCodingOption3));
+	m_co3.Header.BufferId = MFX_EXTBUFF_CODING_OPTION3;
+	m_co3.Header.BufferSz = sizeof(m_co3);
+
+	if (pParams->nbFrames > 1)
+		m_co2.BRefType = MFX_B_REF_PYRAMID;
+
+	PRAGMA_WARN_PUSH
+	PRAGMA_WARN_DEPRECATION
+#pragma warning(disable : 4996)
```

Comment:
```suggestion
```

Define the pragma for MSVC in `libobs/util/c99defs.h`

```c
#define PRAGMA_WARN_DEPRECATION __pragma(warning(disable:4996))
```

Reply:
Fixed this in #9392.